pkg/scheduler/framework/plugins/podtopologyspread/scoring_test.go

```diff
@@ -506,30 +504,91 @@ func BenchmarkTestCalculateEvenPodsSpreadPriority(b *testing.B) {
 	for _, tt := range tests {
 		b.Run(tt.name, func(b *testing.B) {
 			existingPods, allNodes, filteredNodes := st.MakeNodesAndPodsForEvenPodsSpread(tt.pod.Labels, tt.existingPodsNum, tt.allNodesNum, tt.filteredNodesNum)
+			state := framework.NewCycleState()
 			snapshot := nodeinfosnapshot.NewSnapshot(nodeinfosnapshot.CreateNodeInfoMap(existingPods, allNodes))
-			tpSpreadMap, err := buildPodTopologySpreadMap(tt.pod, filteredNodes, snapshot.NodeInfoList)
-			if err != nil {
-				b.Fatal(err)
-			}
-			meta := &priorityMetadata{
-				podTopologySpreadMap: tpSpreadMap,
+			p := &PodTopologySpread{sharedLister: snapshot}
+
+			status := p.PostFilter(context.Background(), state, tt.pod, filteredNodes, nil)
+			if !status.IsSuccess() {
+				b.Fatalf("unexpected error: %v", status)
 			}
 			b.ResetTimer()
 
 			for i := 0; i < b.N; i++ {
 				var gotList framework.NodeScoreList
 				for _, n := range filteredNodes {
 					nodeName := n.Name
-					nodeScore, _ := CalculateEvenPodsSpreadPriorityMap(tt.pod, meta, snapshot.NodeInfoMap[nodeName])
-					gotList = append(gotList, nodeScore)
+					score, status := p.Score(context.Background(), state, tt.pod, nodeName)
+					if !status.IsSuccess() {
+						b.Fatalf("unexpected error: %v", status)
+					}
+					gotList = append(gotList, framework.NodeScore{Name: nodeName, Score: score})
 				}
-				err := CalculateEvenPodsSpreadPriorityReduce(tt.pod, meta, snapshot, gotList)
-				if err != nil {
-					b.Fatal(err)
+
+				status = p.NormalizeScore(context.Background(), state, tt.pod, gotList)
+				if !status.IsSuccess() {
+					b.Fatal(status)
 				}
 			}
 		})
 	}
 }
 
-var softSpread = v1.ScheduleAnyway
+// The tests in this file compare the performance of SelectorSpreadPriority
+// against EvenPodsSpreadPriority with a similar rule.
+
+var (
+	tests = []struct {
+		name            string
+		existingPodsNum int
+		allNodesNum     int
+	}{
+		{
+			name:            "100nodes",
+			existingPodsNum: 1000,
+			allNodesNum:     100,
+		},
+		{
+			name:            "1000nodes",
+			existingPodsNum: 10000,
+			allNodesNum:     1000,
+		},
+	}
+)
+
+func BenchmarkTestDefaultEvenPodsSpreadPriority(b *testing.B) {
+	for _, tt := range tests {
+		b.Run(tt.name, func(b *testing.B) {
+			pod := st.MakePod().Name("p").Label("foo", "").
+				SpreadConstraint(1, v1.LabelHostname, softSpread, st.MakeLabelSelector().Exists("foo").Obj()).
+				SpreadConstraint(1, v1.LabelZoneFailureDomain, softSpread, st.MakeLabelSelector().Exists("foo").Obj()).Obj()
+			existingPods, allNodes, filteredNodes := st.MakeNodesAndPodsForEvenPodsSpread(pod.Labels, tt.existingPodsNum, tt.allNodesNum, tt.allNodesNum)
+			state := framework.NewCycleState()
+			snapshot := nodeinfosnapshot.NewSnapshot(nodeinfosnapshot.CreateNodeInfoMap(existingPods, allNodes))
+			p := &PodTopologySpread{sharedLister: snapshot}
+
+			status := p.PostFilter(context.Background(), state, pod, filteredNodes, nil)
```

Comment:
This is the unfair part to SelectorSpreadPriority.

This calculation should be taken into account in the benchmark.